src/ucs/arch/riscv/global_opts.c

```diff
@@ -0,0 +1,44 @@
+/**
+* Copyright (C) Tactical Computing Labs, LLC. 2022. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#if defined(__riscv)
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucs/arch/global_opts.h>
+#include <ucs/config/parser.h>
+
+ucs_config_field_t ucs_arch_global_opts_table[] = {
+#if ENABLE_BUILTIN_MEMCPY
+  {"BUILTIN_MEMCPY_MIN", "auto",
+   "Minimal threshold of buffer length for using built-in memcpy.",
+   ucs_offsetof(ucs_arch_global_opts_t, builtin_memcpy_min), UCS_CONFIG_TYPE_MEMUNITS},
+
+  {"BUILTIN_MEMCPY_MAX", "auto",
+   "Maximal threshold of buffer length for using built-in memcpy.",
+   ucs_offsetof(ucs_arch_global_opts_t, builtin_memcpy_max), UCS_CONFIG_TYPE_MEMUNITS},
+#endif
+  {NULL}
+};
```

Comment:
are we going to have special memcpy for riscv?

Reply:
A run of `ucx_info`, `ucx_perf`, etc without this call causes illegal instruction segmentation faults. I'd like to retain this code.

Reply:
aarch64 for example does not have it

Reply:
Quick note, there is still a tremendous amount of experimental code and/or lacking features in portions of the RISC-V system software and tool chains. Went back through the patch, updated some `fence.i` locations, and was able to remove this successfully.